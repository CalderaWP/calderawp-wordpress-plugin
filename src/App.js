//import { BrowserRouter as Router, Link } from "react-router-dom";
import React, {Fragment} from 'react';
import Entries from "./Pages/Entries";
import "./app.css";
import {LogoutButton} from "./components/Login/LogoutButton";
import {JwtLogin} from "./components/Login/JwtLogin";
import {Admin} from '@caldera-labs/components'
import { NavigableMenu, TabbableContainer, Button } from '@wordpress/components';
import logo from './globe.svg';
import { createHooks } from '@wordpress/hooks';
export const cfProHooks = createHooks();



class App extends React.Component {

	constructor( props ) {
		super(props);
		this.renderHeader= this.renderHeader.bind(this);
		this.renderBody= this.renderBody.bind(this);
		cfProHooks.addFilter( 'header', 'cfPro', this.renderHeader );
		cfProHooks.addFilter( 'body', 'cfPro', this.renderBody );//mainMenuItems
	};

	renderHeader(){
		const {
			onLogout,
			isLoggedIn
		} = this.props;

		const output =  cfProHooks.applyFilters( 'mainMenuItems', [] );

		return(
			<div id={'caldera-pro-header'}>
				<img src={logo} height={'40px'} width={'40px'} />
				<h1 className="App-title">Caldera Pro </h1>
				<div className={'edit-post-header-toolbar'}>
					<NavigableMenu orientation="horizontal">
						{isLoggedIn &&
							<LogoutButton
								onLogout={onLogout}
							/>
						}
						{output}
					</NavigableMenu>
				</div>
			</div>
		)
	}

	renderBody(){
		const {
			getForms,
			getEntries,
			keys,
			onJwtToken,
			isLoggedIn
		} = this.props;
		return(
			<div id={'caldera-pro-body'}>
				<div className={'caldera-pro-inner'}>
					{isLoggedIn ? (
						<Fragment>
							<Entries
								hooks={cfProHooks}
								getForms={getForms}
								getEntries={getEntries}
							/>
						</Fragment>

					) : (
						<JwtLogin
							onTokenReceived={onJwtToken}
						/>
					)}
				</div>
			</div>
		)
	}
	render() {
		const {
			getForms,
			getEntries,
			keys,
			onJwtToken,
			onLogout,
			isLoggedIn
		} = this.props;

		let output = [];
		output.push( cfProHooks.applyFilters( 'before', [] ) );
		output.push( cfProHooks.applyFilters( 'header', [] ) );
		output.push( cfProHooks.applyFilters( 'body', [] ) );
		output.push( cfProHooks.applyFilters( 'footer', [] ) );

		return (
			<div id="editor" className="gutenberg__editor caldera-pro-editor">
				{ output }
			</div>
		);

	}
}

export default App;


