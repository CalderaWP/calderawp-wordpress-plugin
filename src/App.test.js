import renderer from "react-test-renderer";
import React from "react";
import "@babel/polyfill";
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

import App from './App';

const getForms = jest.fn(() => {
	return Promise.resolve({
		json: jest.fn((r) => {
			return [];
		})
	});
});

const getFormsWithForms = jest.fn(() => {
	return Promise.resolve({
		json: jest.fn((r) => {
			return [
				{
					ID: 'cf1',
					name: 'Form One'
				},
				{
					ID: 'cf2',
					name: 'Form Two'
				}
			];
		})
	});
});

describe('Main app component', () => {
	test('snapshot Logged out', () => {
		expect(renderer.create(<App
			getForms={getForms}
			getEntries={jest.fn()}
			onJwtToken={jest.fn()}
			isLoggedIn={false}
			jwt={''}/>).toJSON()
		).toMatchSnapshot()
	});

	test('snapshot Logged in and no forms', () => {
		expect(renderer.create(<App
			getForms={getFormsWithForms}
			getEntries={jest.fn()}
			onJwtToken={jest.fn()}
			isLoggedIn={true}
			jwt={'asddfghj'}/>).toJSON()
		).toMatchSnapshot()
	});

	test('snapshot Logged in and forms', () => {
		expect(renderer.create(<App
			getForms={getForms}
			getEntries={jest.fn()}
			onJwtToken={jest.fn()}
			isLoggedIn={true}
			jwt={'asddfghj'}/>).toJSON()
		).toMatchSnapshot()
	});
});

describe('Main app component methods', () => {
	const MenuItem = ({name}) => (<div>{name}</div>);

	test('addMenuItem adds a menu item', () => {
		const component = shallow(<App
			getForms={getFormsWithForms}
			getEntries={jest.fn()}
			onJwtToken={jest.fn()}
			isLoggedIn={true}
			jwt={'aa'}/>);

		const item = <MenuItem name={'roy'}/>;
		component.instance().addMenuItem('hello', item);
		expect(component.state('mainMenuItems').hasOwnProperty('hello')).toBe(true);
		expect(component.state('mainMenuItems').hello).toEqual(item);

	});

	test( 'Removing menu items', () => {
		const component = shallow(<App
			getForms={getFormsWithForms}
			getEntries={jest.fn()}
			onJwtToken={jest.fn()}
			isLoggedIn={true}
			jwt={'aa'}/>);

		const item = <MenuItem name={'roy'}/>;
		component.instance().addMenuItem('hello3', <MenuItem name={'Pants!!'}/>);
		component.instance().addMenuItem('hello', <MenuItem name={'Pants!'}/>);
		component.instance().addMenuItem('hello2', item);
	

		component.instance().removeMenuItem('hello' );
		expect(component.state('mainMenuItems').hasOwnProperty('hello')).toBe(false);
		expect(component.state('mainMenuItems').hasOwnProperty('hello2')).toBe(true);
		expect(component.state('mainMenuItems').hasOwnProperty('hello3')).toBe(true);

	});


});

