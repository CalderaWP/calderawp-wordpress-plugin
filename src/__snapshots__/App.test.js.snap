// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main app component snapshot Logged in and forms 1`] = `
<div
  className="gutenberg__editor caldera-pro-editor"
  id="editor"
>
  <div
    id="caldera-pro-header"
  >
    <img
      alt="Caldera Globe Logo"
      height="40px"
      src="globe.svg"
      width="40px"
    />
    <h1
      className="App-title"
    >
      Caldera Pro 
    </h1>
    <div
      className="edit-post-header-toolbar"
    >
      <div
        aria-orientation="horizontal"
        onFocus={undefined}
        onKeyDown={[Function]}
        role="menu"
      >
        <div>
          Hi Roy
        </div>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
    </div>
  </div>
  <div
    id="caldera-pro-body"
  >
    <div
      className="caldera-pro-inner"
    >
      <div>
        <div>
          <label
            className=""
            htmlFor="caldera-forms-form-chooser-entry-viewer-form-chooser-in-app"
          >
            Choose Form
          </label>
          <select
            className="caldera-forms-form-chooser"
            id="caldera-forms-form-chooser-entry-viewer-form-chooser-in-app"
            onChange={[Function]}
            value=""
          >
            <option>
               -- 
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main app component snapshot Logged in and no forms 1`] = `
<div
  className="gutenberg__editor caldera-pro-editor"
  id="editor"
>
  <div
    id="caldera-pro-header"
  >
    <img
      alt="Caldera Globe Logo"
      height="40px"
      src="globe.svg"
      width="40px"
    />
    <h1
      className="App-title"
    >
      Caldera Pro 
    </h1>
    <div
      className="edit-post-header-toolbar"
    >
      <div
        aria-orientation="horizontal"
        onFocus={undefined}
        onKeyDown={[Function]}
        role="menu"
      >
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
    </div>
  </div>
  <div
    id="caldera-pro-body"
  >
    <div
      className="caldera-pro-inner"
    >
      <div>
        <div>
          <label
            className=""
            htmlFor="caldera-forms-form-chooser-entry-viewer-form-chooser-in-app"
          >
            Choose Form
          </label>
          <select
            className="caldera-forms-form-chooser"
            id="caldera-forms-form-chooser-entry-viewer-form-chooser-in-app"
            onChange={[Function]}
            value=""
          >
            <option>
               -- 
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main app component snapshot Logged out 1`] = `
<div
  className="gutenberg__editor caldera-pro-editor"
  id="editor"
>
  <div
    id="caldera-pro-header"
  >
    <img
      alt="Caldera Globe Logo"
      height="40px"
      src="globe.svg"
      width="40px"
    />
    <h1
      className="App-title"
    >
      Caldera Pro 
    </h1>
    <div
      className="edit-post-header-toolbar"
    >
      <div
        aria-orientation="horizontal"
        onFocus={undefined}
        onKeyDown={[Function]}
        role="menu"
      />
    </div>
  </div>
  <div
    id="caldera-pro-body"
  >
    <div
      className="caldera-pro-inner"
    >
      <form
        id="caldera-pro-jwt-login"
        onSubmit={[Function]}
      >
        <div>
          <div
            className="alert"
          >
            
          </div>
        </div>
        <div
          className="components-base-control"
        >
          <div
            className="components-base-control__field"
          >
            <label
              className="components-base-control__label"
              htmlFor="inspector-text-control-0"
            >
              User name
            </label>
            <input
              aria-describedby={undefined}
              className="components-text-control__input"
              disabled={false}
              id="caldera-pro-cfp-username"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="components-base-control"
        >
          <div
            className="components-base-control__field"
          >
            <label
              className="components-base-control__label"
              htmlFor="inspector-text-control-1"
            >
              Password
            </label>
            <input
              aria-describedby={undefined}
              className="components-text-control__input"
              disabled={false}
              id="caldera-pro-cfp-password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
        </div>
        <input
          id="caldera-pro-jwt-login-submit"
          type="submit"
          value="Login"
        />
      </form>
    </div>
  </div>
</div>
`;
